client/src/util/stateManager/matrix.ts

```diff
@@ -242,17 +283,24 @@ export function matrixFBSToDataframe(
   const columns = fbs
     .map((fb) =>
       fb.columns.map((c) => {
-        if (isFloatTypedArray(c) || Array.isArray(c)) return c;
+        if (isFloatTypedArray(c) || isCatTypedArray(c) || Array.isArray(c))
+          return c;
         return promoteTypedArray(c);
       })
     )
     .flat();
+
   // colIdx may be TypedArray or Array
   const colIdx = fbs
     .map((b) => (Array.isArray(b.colIdx) ? b.colIdx : Array.from(b.colIdx)))
     .flat();
   const nCols = columns.length;
-
   const df = new Dataframe([nRows, nCols], columns, null, new KeyIndex(colIdx));
   return df;
 }
+
+export const isCatTypedArray = (c: unknown): boolean =>
```

Comment:
wrong place for this code - it is general purpose and not specific to FB serialization.  Doesn't it belong in arraytypes.ts or something like that?

Reply:
Ran into some annoying circular dependencies...